app/controllers/reviews_controller.rb

```diff
@@ -27,6 +27,31 @@ def create
     end
   end
 
+  def edit
+    @review = Review.find(params[:id])
+    if request.xhr?
+      render partial: "reviews/edit"
+    end
+  end
+
+  def update
+    @review = Review.find(params[:id])
+    if @review.update_attributes(review_params)
+      if request.xhr?
+        flash[:notice] = 'Review has been updated'
+        render :js => "window.location = '#{review_path(params[:id])}';"
```

Comment:
You probably have to empty and append the flash notice in the rendered JS directly for this branch: 
```Javascript
window.location = '#{review_path(params[:id])}';$('.alert-notice').empty().append('<p>Review has been updated</p><br />');